tests/edit_environment.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+#
+# This script injects proxy, DNS, and NTP configurations into the
+# "Test-Laptop" environment for VM cluster builds.
+#
+# We attempt to auto-detect these values.  To override DNS, NTP, or
+# proxy servers, export the appropriate environment variable before
+# calling ./tests/automated_install.sh.
+#
+# Proxy:
+#   http_proxy
+#
+# DNS server:
+#   BACH_DNS_SERVER
+#
+# NTP server:
+#   BACH_NTP_SERVER
+#
+
+require 'json'
+require 'uri'
+
+def get_name_server
+  ns = File.readlines('/etc/resolv.conf').select do |ll|
+    ll.include?('nameserver')
+  end.first.chomp.gsub(/^\s*nameserver\s*/,'')
+
+  if ns == '127.0.0.1'
+    # Look to see if we have dhclient running
+    leases =
+      Dir.glob('/var/lib/dhcp/dhclient*.lease').map do |f|
+        [File.mtime(f), f]
+      end.flatten.to_h
+    latest = leases.keys.sort.last or return nil
+    ns = File.readlines(leases[latest]).select do |ll|
+      ll.include?('option domain-name-servers')
+    end.first.chomp.gsub(/^\s*option domain-name-servers\s*([^,]*),.*/,'\1')
+  else
+    ns
+  end
+end
+
+name_server = ENV['BACH_DNS_SERVER'] || get_name_server
+ntp_server = ENV['BACH_NTP_SERVER'] || name_server
+proxy_uri = if ENV['http_proxy']
+              URI.parse(ENV['http_proxy'])
+            else
+              nil
+            end
+
+repo_dir = File.expand_path(File.join(File.dirname(__FILE__), '..'))
+
+#
+# Some versions of VBox were fond of running a DHCP server by default
+# on every virtual network.  Remove it, if present.
+#
+system('vboxmanage dhcpserver remove ' +
+       '--netname HostInterfaceNetworking-vboxnet0')
+
+#
+# Remove any prior local changes made by shell scripts
+#
+Dir.chdir(repo_dir) {
+  system('git checkout -- Vagrantfile')
+  system('rm -f Berksfile.lock')
```

Comment:
`../cluster` is left alone on purpose, because I commonly make changes to the environment, roles, or cluster.txt.

I don't understand what you mean my tracking changes in stub-environment.